packages/httpsnippet-client-api/src/index.js

```diff
@@ -53,7 +52,12 @@ function getAuthSources(operation) {
         if (scheme.in === 'query') {
           matchers.query.push(scheme.name);
         } else if (scheme.in === 'header') {
-          matchers.header[scheme.name] = '*';
+          // The way that this astrisk header matcher works is that since this `apiKey` goes in a
+          // named header (`scheme.name`) because the header is the key, we're matching against the
+          // entire header -- counter to the way that the HTTP basic matcher above works where we
+          // match and extract the API key from everything after `Basic ` in the `Authorization`
+          // header.
+          matchers.header[scheme.name.toLowerCase()] = '*';
```

Comment:
Hope this comment makes sense because it took me a good bit staring this line and how it's used further below before understanding why I wrote it this way.